tests/client_api.cpp

```diff
@@ -107,6 +107,107 @@ TEST(ClientAPI, CreateRoom)
+                // Block while we are logging in
+                continue;
+        }
+        mtx_client->set_access_token(token);
+        mtx::requests::CreateRoom req;
+        req.name  = "Test1";
+        req.topic = "Topic1";
+        mtx_client->create_room(req, [](const mtx::responses::CreateRoom &res, ErrType err) {
+                boost::ignore_unused(res);
+                ASSERT_FALSE(err);
+        });
+
+        // Logout and prove that logout was successful and deleted the access_token_ for the client
+        mtx_client->logout([mtx_client, &token](const mtx::responses::Logout &res, ErrType err) {
+                boost::ignore_unused(res);
+                ASSERT_FALSE(err);
+                token.clear();
+        });
+        while (token.size()) {
+                // Block while we are logging out
+                continue;
+        }
+        // Verify that sending requests with this mtx_client fails after logout
+        mtx::requests::CreateRoom failReq;
+        failReq.name  = "42";
+        failReq.topic = "LifeUniverseEverything";
+        mtx_client->create_room(failReq, [](const mtx::responses::CreateRoom &res, ErrType err) {
+                boost::ignore_unused(res);
+                ASSERT_TRUE(err);
+                EXPECT_EQ(mtx::errors::to_string(err->matrix_error.errcode), "M_MISSING_TOKEN");
+                EXPECT_EQ(err->status_code, boost::beast::http::status::forbidden);
+
+        });
+
+        mtx_client->close();
+}
+
+TEST(ClientAPI, LogoutInvalidatesTokenOnServer)
+{
+        std::shared_ptr<Client> mtx_client = std::make_shared<Client>("localhost");
+        std::string token;
+
+        // Login and prove that login was successful by creating a room
+        mtx_client->login(
+          "alice", "secret", [&token](const mtx::responses::Login &res, ErrType err) {
+                  ASSERT_FALSE(err);
+                  validate_login("@alice:localhost", res);
+                  token = res.access_token;
+          });
+        while (token.empty()) {
+                // Block while we are logging in
+                continue;
+        }
+        mtx_client->set_access_token(token);
+        mtx::requests::CreateRoom req;
+        req.name  = "Test1";
+        req.topic = "Topic1";
+        mtx_client->create_room(req, [](const mtx::responses::CreateRoom &res, ErrType err) {
+                boost::ignore_unused(res);
+                ASSERT_FALSE(err);
+        });
+
+        // Logout and prove that logout was successful by verifying the old access_token_ is no
+        // longer valid
+        mtx_client->logout([mtx_client, &token](const mtx::responses::Logout &res, ErrType err) {
+                boost::ignore_unused(res);
+                ASSERT_FALSE(err);
+                mtx_client->set_access_token(token);
+                token.clear();
+        });
+        while (token.size()) {
+                // Block while we are logging out
+                continue;
```

Comment:
Could you replace the `continue` statement with `std::this_thread::sleep_for` like in the previous test cases, because currently it uses unnecessarily cpu time. Something between 100 & 500 msecs would do.

Reply:
Good point, thanks! I just added it in now.